src/bootstrap/src/core/builder/tests.rs

```diff
@@ -1233,3 +1234,29 @@ fn any_debug() {
     // Downcasting to the underlying type should succeed.
     assert_eq!(x.downcast_ref::<MyStruct>(), Some(&MyStruct { x: 7 }));
 }
+
+/// Check that during a non-cross-compiling stage 2 build, we only compile rustc host tools
+/// (such as llvm-bitcode-linker) only once.
+#[test]
+fn llvm_bitcode_linker_compile_once() {
```

Comment:
Question: should we have another test that *does* check the cross-compiling case, that we *do* build a stage 1 (rustc, std) pair that can produce artifacts for `target`, and that `llvm-bitcode-linker` is produced by that pair for the `target`?